src/Pillar-Tests-Cli/ClapPillarBuildCommandTest.class.st

```diff
@@ -0,0 +1,34 @@
+Class {
+	#name : #ClapPillarBuildCommandTest,
+	#superclass : #TestCase,
+	#category : #'Pillar-Tests-Cli'
+}
+
+{ #category : #adding }
+ClapPillarBuildCommandTest >> testAddSubcommand [
+	| allIds defaultCommand |
+	defaultCommand := ClapPillarBuildCommand commandSpecification.
+	allIds := PRTarget clapCommandList collect: [ :c | c identifier ].
+	allIds do: 
+		[ :id | 
+			self assert: 
+				(defaultCommand subcommands anySatisfy: 
+					[ :c | c class = ClapCommand new class and: c identifier = id asSymbol]
+			) 
```

Comment:
1. Just compare the object.

The problem is that you're extracting the ids before in
```smalltalk
allIds := PRTarget clapCommandList collect: [ :c | c identifier ].
```

So then you only have the ID to compare!
Instead, if you get the command list
```smalltalk
allCommands := PRTarget clapCommandList.
```

and knowing that `aCollection anySatisfy: [:element | element = anotherObject ]` is the same as `aCollection includes: anotherObject`. Then you could do:

```smalltalk
defaultCommand subcommands do: [ :sc | allCommands includes: sc ].
```
Of course, this means we have to define `=`and `hash` in commands.
@cdlm is equality defined? and if yes, how?


2. `ClapCommand new class` is the same as just writing `ClapCommand`

Reply:
Equality is defined by `Object` (so, identity).